app/src/influxdb_writer.py

```
@@ -0,0 +1,319 @@
+        stop = date.replace(hour=23, minute=59, second=59, microsecond=999999)
+
+        # Escape single quotes to keep share_symbol a safe SQL string literal
+        safe_symbol = share_symbol.replace("'", "''")
+        # Use SQL query for InfluxDB 3
+        query = f"""
+        SELECT COUNT(*) as count
+        FROM "{self.MEASUREMENT}"
+        WHERE share_symbol = '{safe_symbol}'
+          AND time >= '{start.isoformat()}Z'
+          AND time <= '{stop.isoformat()}Z'
+        """
```

Ruff (0.15.20)

[error] Possible SQL injection vector through string-based query construction

(S608)